### DISCUSSION TITLE
updated to 11.1.2 and the browser project does not build because it does not find the namespace avalonia.browser

### DISCUSSION THREAD

**Discussion by simsfarrow** [AvaloniaUI/Avalonia#16634] [Q&A] [answered]
### Describe the bug

<PackageReference Include="Avalonia.Browser" Version="11.1.2" />

using Avalonia.Browser; <- The namespace does not exist in the namespace avalonia.

### To Reproduce

Upgrade to the latest version

### Avalonia version

11.1.2

### OS

WebAssembly

**Comment by maxkatz6** [comment]
@simsfarrow can you provide a repro? Freshly created project from the templates builds.

**Comment by maxkatz6** [comment]
Actually, if you are migrating from previous version, you might be missing `net8.0-browser` target framework, as we dropped older browser and mobile targets in 11.1.

In your csproj
```
<TargetFramework>net8.0-browser</TargetFramework>
```

**Comment by simsfarrow** [comment]
The project prior to upgrade worked and was 

<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <RuntimeIdentifier>browser-wasm</RuntimeIdentifier>
    <WasmMainJSPath>AppBundle\main.js</WasmMainJSPath>
    <OutputType>Exe</OutputType>
  </PropertyGroup>

  <ItemGroup>
    <WasmExtraFilesToDeploy Include="AppBundle\**" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Avalonia.Browser" Version="11.0.10" />
  </ItemGroup>

Changing to Version below caused the error.

<PackageReference Include="Avalonia.Browser" Version="11.1.2" />


I FIXED IT by creating a fresh project and brought across the wwwroot folder from the fresh project and deleted AppBundle folder and changed the project to

Project Sdk="Microsoft.NET.Sdk.WebAssembly">
	<PropertyGroup>
		<TargetFramework>net8.0-browser</TargetFramework>
		<OutputType>Exe</OutputType>
		<AllowUnsafeBlocks>true</AllowUnsafeBlocks>
	</PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Avalonia.Browser" Version="11.1.2" />
  </ItemGroup>

**Reply by simsfarrow** [reply, marked as answer]
The project prior to upgrade worked and was

Project Sdk="Microsoft.NET.Sdk"
  PropertyGroup
    TargetFrameworknet8.0TargetFramework
    RuntimeIdentifierbrowser-wasmRuntimeIdentifier
    WasmMainJSPathAppBundle\main.jsWasmMainJSPath
    OutputTypeExeOutputType
  PropertyGroup

  ItemGroup
    WasmExtraFilesToDeploy Include="AppBundle\**" 
  ItemGroup

  ItemGroup
    PackageReference Include="Avalonia.Browser" Version="11.0.10" 
  ItemGroup


Changing to Version below caused the error.

 PackageReference Include="Avalonia.Browser" Version="11.1.2" 

I FIXED IT by creating a fresh project and brought across the wwwroot folder from the fresh project and deleted AppBundle folder and changed the project to

Project Sdk="Microsoft.NET.Sdk.WebAssembly"
	PropertyGroup
		TargetFrameworknet8.0-browserTargetFramework
		OutputTypeExeOutputType
		AllowUnsafeBlockstrueAllowUnsafeBlocks
	PropertyGroup

  ItemGroup
    PackageReference Include="Avalonia.Browser" Version="11.1.2" 
  ItemGroup
